backend/src/reservations/reservations.service.ts

```diff
@@ -31,53 +31,56 @@ export class ReservationService {
       if (sendSms) {
         const smsResponse = await this.sendSmsToUser(reservation);
         if (smsResponse) {
-          await Promise.all(
-            reservation.tickets.map(
-              async ticket =>
-                await this.priceService.increaseOccupiedSeats(
-                  ticket.price_id,
-                  ticket.no_of_tickets,
-                ),
-            ),
-          );
-          this.updateReservationStatus(reservation.id, true);
-          return response;
-        } else {
-          console.error('Sms Funtionality failed!.');
+          await this.updateReservation(reservation.id, { sms_sent: true })
         }
-      } else {
-        return response;
       }
+      await Promise.all(
+        reservation.tickets.map(
+          async ticket =>
+            await this.priceService.updateSeats(
+              ticket.price_id,
+              ticket.no_of_tickets,
+            ),
+        ),
+      );
+      return response;
+
     } else {
       throw new Error('Failed to create reservation!.');
     }
   }
 
   async deleteReservation(id: number) {
+    const reservation = await this.findOneById(id);
+    await Promise.all(
+      reservation.tickets.map(
+        async ticket =>
+          await this.priceService.updateSeats(
```

Comment:
isnt it that you return an array of promises? then you dont await anymore.
Sync func, return promise

Reply:
I am using await Promise.all(). Isnt it supposed to wait till the promise resolves here.

Reply:
Promise.all do it behind the scene, you dont explitcitly wait here